qa/L0_perf_resnet/run_test.sh

```diff
@@ -73,22 +66,41 @@ fi
 
 set +e
 
-# Run the model once to warm up. Some frameworks do optimization on the first requests.
-# Must warmup similar to actual run so that all instances are ready
-$PERF_CLIENT -v -i ${PERF_CLIENT_PROTOCOL} -m $MODEL_NAME -p${MEASUREMENT_WINDOW} \
-                -b${STATIC_BATCH} --concurrency-range ${CONCURRENCY}
+# Overload use of PERF_CLIENT_PROTOCOL for convenience with existing test and 
+# reporting structure, though "triton_c_api" is not strictly a "protocol".
+if [[ "${PERF_CLIENT_PROTOCOL}" == "triton_c_api" ]]; then
+    # Server will be run in-process with C API
+    SERVICE_ARGS="--service-kind triton_c_api \
+                  --triton-server-directory ${TRITON_DIR} \
+                  --model-repository ${MODEL_REPO}"
+else
+    SERVICE_ARGS="-i ${PERF_CLIENT_PROTOCOL}"
+
+    SERVER_LOG="${NAME}.serverlog"
+    run_server
+    if (( $SERVER_PID == 0 )); then
+        echo -e "\n***\n*** Failed to start $SERVER\n***"
+        cat $SERVER_LOG
+        exit 1
+    fi
+
+    # Run the model once to warm up. Some frameworks do optimization on the first requests.
+    # Must warmup similar to actual run so that all instances are ready
+    # Note: Running extra PA for warmup doesn't make sense for C API since it
+    # uses in-process tritonserver which will exit along with this PA process.
+    $PERF_CLIENT -v -m $MODEL_NAME -p${MEASUREMENT_WINDOW} \
+                    -b${STATIC_BATCH} --concurrency-range ${CONCURRENCY} \
+                    ${SERVICE_ARGS}
+fi
 
-$PERF_CLIENT -v -i ${PERF_CLIENT_PROTOCOL} -m $MODEL_NAME -p${MEASUREMENT_WINDOW} \
+# Measure perf client results and write them to a file for reporting
+$PERF_CLIENT -v -m $MODEL_NAME -p${MEASUREMENT_WINDOW} \
                 -b${STATIC_BATCH} --concurrency-range ${CONCURRENCY} \
+                ${SERVICE_ARGS} \
                 -f ${NAME}.csv 2>&1 | tee ${NAME}.log
 if (( $? != 0 )); then
     RET=1
 fi
-curl localhost:8002/metrics -o ${NAME}.metrics >> ${NAME}.log 2>&1
```

Comment:
@tanmayv25 @Tabrizian @GuanLuo What do you think of removing this? It does not appear to be used anywhere. The metrics aren't pushed to kibana and aren't read in [reporter.py](https://github.com/triton-inference-server/server/blob/main/qa/common/reporter.py)

Reply:
Based on Tanmay's comment: https://github.com/triton-inference-server/server/pull/4420#discussion_r881063460

Reply:
I think it should be fine to remove them if it is not being used anywhere. The PA report already includes the server stats if we decide to use them. The only metrics that are not available is the GPU usage which I don't think we're tracking them anywhere.